api/v1/controllers/AuthController.js

```diff
@@ -0,0 +1,68 @@
+var Promise = require('bluebird');
+
+var errors = require('../errors');
+var AuthService = require('../services/AuthService');
+var UserService = require('../services/UserService');
+
+var router = require('express').Router();
+
+function _issueByEmail(email, password) {
+	if (!password && password !== null) {
+		throw new TypeError("To issue a token by email without checking the " +
+			"user's password, you must pass the password as the null type");
```

Comment:
Good idea for devs working in the function, an optional argument should do the same thing while not confusing external consumers   
  
  

  
  
>   
> On Jun 23, 2016 at 2:57 PM,  <Nick Magerko (mailto:notifications@github.com)>  wrote:
>   
>   
>   
>
> In   api/v1/controllers/AuthController.js (https://github.com/HackIllinois/api-2017/pull/1#discussion_r68313601):
>
>   
>   >  @@ -0,0 +1,68 @@  >  +var Promise = require('bluebird');  >  +  >  +var errors = require('../errors');  >  +var AuthService = require('../services/AuthService');  >  +var UserService = require('../services/UserService');  >  +  >  +var router = require('express').Router();  >  +  >  +function _issueByEmail(email, password) {  >  + if (!password  &&  password !== null) {  >  + throw new TypeError("To issue a token by email without checking the " +  >  + "user's password, you must pass the password as the null type");  
>   
>
> The reason I forced a  null  argument here is so that people who are working with this underlying service method are conscious of the fact that they can (and sometimes should) be checking the user's password before issuing a token.
>
>   
>
> I guess that might be a bit of an overbearing precaution though -- I'll remove it.
>
>   
>
> —
>  You are receiving this because you commented.
>  Reply to this email directly,   view it on GitHub (https://github.com/HackIllinois/api-2017/pull/1/files/b88df4dbc2a31758d6a8a0ba08cef9838e75c7c2#r68313601), or   mute the thread (https://github.com/notifications/unsubscribe/ABtSlaBfqtOQyq9nvl_i7bGvRqqRsBPXks5qOvMhgaJpZM4I5F0q).
>
>   
>   
>     
>   
>   
>